plugins/web/opentelemetry-plugin-user-interaction/src/userInteraction.ts

```diff
@@ -313,28 +313,31 @@ export class UserInteractionPlugin extends BasePlugin<unknown> {
       ): Zone {
         const target: HTMLElement | undefined = plugin._getClickedElement(task);
         let span: api.Span | undefined;
+        const activeZone = this;
         if (target) {
           span = plugin._createSpan(target, 'click');
           if (span) {
             plugin._incrementTask(span);
-            try {
-              return plugin._tracer.withSpan(span, () => {
-                const currentZone = Zone.current;
-                task._zone = currentZone;
-                return original.call(currentZone, task, applyThis, applyArgs);
-              });
-            } finally {
-              plugin._decrementTask(span);
-            }
+            return activeZone.run(() => {
+              try {
+                return plugin._tracer.withSpan(span as api.Span, () => {
```

Comment:
Why do you have to do Type Assertion here (`span as api.Span`)?

Reply:
should not be required as the assertion on line 319 should remove `undefined` from the type

Reply:
this is a compiler error / issue. the internal _createSpan returns api.Span or undefined. Even though there is `if` check for span existence the compiler still complains. If you have a better idea how to change it I'm open for suggestion.